### PR TITLE
moving Quantity private doctests to pytests.

### PR DESCRIPTION
**Pull request by mmatera**
This part required some changes in `mathics.builtin.quantity`, because the original one was not compatible with MathicsSession evaluation.

### CODE REVIEW

File: mathics/builtin/quantities.py [hunk 1]
```diff
@@ -185,88 +175,63 @@ class QuantityMagnitude(Builtin):
...
+    def eval_List2(self, expr, unit, evaluation: Evaluation):
+        "QuantityMagnitude[expr_List, unit_]"
+        return ListExpression(
+            *(
+                Expression(Symbol(self.get_name()), e, unit).evaluate(evaluation)
+                for e in expr.elements
+            )
+        )
 
-        def get_magnitude(elements):
```
Comment (mmatera) [inline mathics/builtin/quantities.py:214]: Here I tried to disentangle this by adding specific methods for each different form.

File: mathics/builtin/quantities.py [hunk 2]
```diff
@@ -185,88 +175,63 @@ class QuantityMagnitude(Builtin):
...
-    #> QuantityMagnitude[Quantity[3,"mater"]]
-     : Unable to interpret unit specification mater.
-     = QuantityMagnitude[Quantity[3,mater]]
     """
 
     summary_text = "get magnitude associated with a quantity."
 
-    def eval(self, expr, evaluation: Evaluation):
-        "QuantityMagnitude[expr_]"
+    def eval_List1(self, expr, evaluation: Evaluation):
```
Comment (rocky) [inline mathics/builtin/quantities.py:182]: This and `eval_list2 ()` and `validate_unit` should probably get moved to the `mathics.eval.quantities` module.

**Comment by rocky** [comment]
LGTM - see comment should you want to address this round.

File: mathics/builtin/quantities.py [hunk 3]
```diff
@@ -31,16 +34,30 @@
...
+    # TODO: This looks very hacky. Reformulate in a clearer
+    # way.
     magnitude_elements = list(magnitude_expr.elements)
     magnitude_elements[1] = from_python(magnitude_elements[1])
     magnitude_expr._elements = tuple(magnitude_elements)
     # FIXME: consider returning an int when that is possible
     return magnitude_expr.evaluate(evaluation).get_float_value()
 
 
+def validate_unit(unit):
```
[the remark this answers is not in the document]
Comment (rocky) [reply mathics/builtin/quantities.py]: I really appreciate the subsequent function renaming.

**Pull request merged**